pkg/cloud/aws/aws.go

```diff
@@ -85,18 +65,22 @@ func RunOnAWS(restConfigProducer restconfig.Producer, gwInstanceType string,
 
 	reporter.Succeeded("")
 
-	k8sConfig, err := restConfigProducer.ForCluster()
-	exit.OnErrorWithMessage(err, "Failed to initialize a Kubernetes config")
-
-	restMapper, err := util.BuildRestMapper(k8sConfig)
-	exit.OnErrorWithMessage(err, "Failed to create restmapper")
+	restMapper, err := util.BuildRestMapper(config)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to create restmapper")
+	}
 
-	dynamicClient, err := dynamic.NewForConfig(k8sConfig)
-	exit.OnErrorWithMessage(err, "Failed to create dynamic client")
+	dynamicClient, err := dynamic.NewForConfig(config)
```

Comment:
Not using `clientProducer` because only `dynamicClient` is used in the code.

Reply:
I thought the general idea was to standardise on `clientProducer` and pass that around, regardless of what is actually used.

Reply:
We shouldn't pass in `rest.Config` to lib functions. If a function really only needs a dynamic or kube client, I think it's OK to just pass that instead of the heavier `client.Producer`. We've done that in other places.